napari/_qt/layer_controls/qt_shapes_controls.py

```diff
@@ -409,25 +426,12 @@ def _on_current_face_color_change(self):
         with qt_signals_blocked(self.faceColorEdit):
             self.faceColorEdit.setColor(self.layer.current_face_color)
 
-    def _on_editable_change(self):
-        """Receive layer model editable change event & enable/disable buttons."""
+    def _on_editable_or_visible_change(self):
```

Comment:
We could consider defining this on the base layer controls widget and defining `_EDIT_BUTTONS` or maybe `_get_edit_buttons()` on that base layer and overriding on others. But I resisted doing that for now to avoid this PR growing too large and away from its initial intent.

Reply:
haha as I was reading the PR I had the exact same thought. Happy to punt and make an issue.